### PR TITLE
Fix two tests with complex PETSc

### PR DESCRIPTION
**Pull request by kronbichler**
Note that I do not have a complex PETSc version around locally, so this should be taken with a grain of salt.

### CODE REVIEW

**Comment by kronbichler** [comment]
Test failures see https://cdash.dealii.43-1.org/test/14719224 and https://cdash.dealii.43-1.org/test/14721238

**Review by tamiko — APPROVED** [review]

**Comment by tamiko** [comment]
@kronbichler I will check this real quick on the tester.

File: tests/petsc_complex/assemble_01.cc [hunk 1]
```diff
@@ -37,9 +37,20 @@ main(int argc, char **argv)
 
   const unsigned int n = 4;
 
-  PETScWrappers::SparseMatrix      serial_matrix(n, n, n);
-  PETScWrappers::MPI::SparseMatrix mpi_matrix(MPI_COMM_WORLD, n, n, n, n, n);
-  PETScWrappers::MPI::Vector       mpi_vector(MPI_COMM_WORLD, n, n);
+  PETScWrappers::SparseMatrix serial_matrix(n, n, n);
+
+  // PETSc requires us to set up the sparsity pattern ahead of construction
+  DynamicSparsityPattern dsp(n, n);
```
Comment (tamiko) [inline tests/petsc_complex/assemble_01.cc:43]:
```
diff --git a/tests/petsc_complex/assemble_01.cc b/tests/petsc_complex/assemble_01.cc
index b055c8a93b..e31b5a3c49 100644
--- a/tests/petsc_complex/assemble_01.cc
+++ b/tests/petsc_complex/assemble_01.cc
@@ -19,6 +19,7 @@
 // combinations of PETScWrappers objects is correctly instantiated and works.
 
 #include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
 #include <deal.II/lac/full_matrix.h>
 #include <deal.II/lac/petsc_sparse_matrix.h>
 #include <deal.II/lac/petsc_vector.h>
```

**Review by tamiko — CHANGES_REQUESTED** [review]

File: tests/petsc_complex/assemble_01.cc (hunk 1 above)
Comment (kronbichler) [reply tests/petsc_complex/assemble_01.cc:44]: Thank you!

**Pull request merged**